tianshou/highlevel/experiment.py

```diff
@@ -0,0 +1,1099 @@
+        hidden_activation: ModuleType = torch.nn.ReLU,
+    ) -> Self:
+        """Makes both critics use the default, MLP-style architecture with the given parameters.
+
+        :param hidden_sizes: the sequence of dimensions to use in hidden layers of the network
+        :param hidden_activation: the activation function to use for hidden layers
+        :return: the builder
+        """
+        for i in range(len(self._critic_factories)):
+            self._with_critic_factory_default(i, hidden_sizes, hidden_activation)
+        return self
+
+    def with_common_critic_factory_use_actor(self) -> Self:
+        """Makes both critics reuse the actor's preprocessing network (parameter sharing)."""
+        for i in range(len(self._critic_factories)):
+            self._with_critic_factory_use_actor(i)
+        return self
+
+    def with_critic1_factory(self, critic_factory: CriticFactory) -> Self:
+        """Specifies that the given factory shall be used for the first critic.
+
+        :param critic_factory: the critic factory
+        :return: the builder
+        """
+        self._with_critic_factory(0, critic_factory)
+        return self
+
+    def with_critic1_factory_default(
+        self,
+        hidden_sizes: Sequence[int] = CriticFactoryDefault.DEFAULT_HIDDEN_SIZES,
+        hidden_activation: ModuleType = torch.nn.ReLU,
+    ) -> Self:
+        """Makes the first critic use the default, MLP-style architecture with the given parameters.
+
+        :param hidden_sizes: the sequence of dimensions to use in hidden layers of the network
+        :param hidden_activation: the activation function to use for hidden layers
+        :return: the builder
+        """
+        self._with_critic_factory_default(0, hidden_sizes, hidden_activation)
+        return self
+
+    def with_critic1_factory_use_actor(self) -> Self:
+        """Makes the first critic reuse the actor's preprocessing network (parameter sharing)."""
+        return self._with_critic_factory_use_actor(0)
+
+    def with_critic2_factory(self, critic_factory: CriticFactory) -> Self:
+        """Specifies that the given factory shall be used for the second critic.
+
+        :param critic_factory: the critic factory
+        :return: the builder
+        """
+        self._with_critic_factory(1, critic_factory)
+        return self
+
+    def with_critic2_factory_default(
+        self,
+        hidden_sizes: Sequence[int] = CriticFactoryDefault.DEFAULT_HIDDEN_SIZES,
+        hidden_activation: ModuleType = torch.nn.ReLU,
+    ) -> Self:
+        """Makes the second critic use the default, MLP-style architecture with the given parameters.
+
+        :param hidden_sizes: the sequence of dimensions to use in hidden layers of the network
+        :param hidden_activation: the activation function to use for hidden layers
+        :return: the builder
+        """
+        self._with_critic_factory_default(0, hidden_sizes, hidden_activation)
```

Comment:
```suggestion
        self._with_critic_factory_default(1, hidden_sizes, hidden_activation)
```

Reply:
Please double check other cases, I just came across to this bug

Reply:
Good catch. Found no further index-related issues.